Tests/tests/test_HdmiCec.cpp

```diff
@@ -0,0 +1,258 @@
+
+}
+
+TEST_F(HdmiCecInitializedEventTest, cecAddressesChanged)
+{
+    EXPECT_CALL(libCCECImplMock, getPhysicalAddress(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](uint32_t *physAddress) {
+                *physAddress = (uint32_t)1234;
+            }));
+    ASSERT_TRUE(dsHdmiEventHandler != nullptr);
+    EXPECT_CALL(service, Submit(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](const uint32_t, const Core::ProxyType<Core::JSON::IElement>& json) {
+                string text;
+                EXPECT_TRUE(json->ToString(text));
+                EXPECT_EQ(text, string(_T("{\"jsonrpc\":\"2.0\",\"method\":\"org.rdk.HdmiCec.cecAddressesChanged\",\"params\":{\"CECAddresses\":{\"physicalAddress\":1234}}}")));
+
+                return Core::ERROR_NONE;
+            }));
+    IARM_Bus_DSMgr_EventData_t eventData;
+    eventData.data.hdmi_hpd.event = 0;	
+    handler.Subscribe(0, _T("cecAddressesChanged"), _T("org.rdk.HdmiCec"), message);
+    dsHdmiEventHandler(IARM_BUS_DSMGR_NAME, IARM_BUS_DSMGR_EVENT_HDMI_HOTPLUG, &eventData , 0);
+    handler.Unsubscribe(0, _T("cecAddressesChanged"), _T("org.rdk.HdmiCec"), message);
+}
+
+TEST_F(HdmiCecInitializedTest, getDeviceList)
+{  
+    int iCounter = 0;
+    //Checking to see if one of the values has been filled in (as the rest get filled in at the same time, and waiting if its not.
+    while ((!Plugin::HdmiCec::_instance->deviceList[0].m_isOSDNameUpdated) && (iCounter < (2*10))) { //sleep for 2sec.
+		usleep (100 * 1000); //sleep for 100 milli sec
+		iCounter ++;
+	}
+
+    //Calling the device list, which is a defualt list of the hdmiCec class. Kist grabs the deviceList.
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getDeviceList"), _T(""), response));
+    //m_osdName and m_vendorID do not get populated in the population function, like logical address, and thus are empty with actual implementtion of .toString()
+    EXPECT_EQ(response, string(_T("{\"numberofdevices\":14,\"deviceList\":[{\"logicalAddress\":1,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":2,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":3,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":4,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":5,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":6,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":7,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":8,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":9,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":10,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":11,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":12,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":13,\"osdName\":\"\",\"vendorID\":\"\"},{\"logicalAddress\":14,\"osdName\":\"\",\"vendorID\":\"\"}],\"success\":true}")));
+
+
+}
```

Comment:
It is not clear what you test in getDeviceList. It's a black box with functionality unknon to the reader. It is not clear what's the setup - why sleep is needed. It is not clear what are the magic values returned, where they come from, are they valid/not valid. If there's no good test, don't test please.

Reply:
@npoltorapavlo 

The reason the sleep is needed is that we're dealing with threads. HDMICec is not designed to be booted up, and immediately try and grab variables that are assigned in the the threads that get spun of on it's creation. The sleep is in there, is such that we try and wait until something gets populated (ideally the entire array of 16), before calling the the function the gets deviceList. Without a sleep, we will get a random number of devices, anywhere from 0-16, depending on how long errant processes took.

As for what we are testing, this is the important part of the getDeviceListFunction
```
int i = 0;
			for(i=0; i< LogicalAddress::UNREGISTERED; i++ ) {
				if (BIT_CHECK(deviceList[i].m_deviceInfoStatus, BIT_DEVICE_PRESENT)) {
					JsonObject device;
					device["logicalAddress"] = HdmiCec::_instance->deviceList[i].m_logicalAddress.toInt();
					device["osdName"] = HdmiCec::_instance->deviceList[i].m_osdName.toString().c_str();
					device["vendorID"] = HdmiCec::_instance->deviceList[i].m_vendorID.toString().c_str();
					deviceListArg.Add(device);
				}
			}
```

Earlier, since the implementation was simple, you told me to use the actual implementation of the simple functions (toString, to toInt, etc), rather then mocking and explicitly saying what values to return. Thus these are the values that those actual implementations return in this instance.

I can do two things, change the actual implementation for toString and toInt back to the mocked version, such that I can specify the returns , and check specifically for those values (which does work, as that was the implementation I had before the change), or I remove it, which I don't really want to do. I know what we're testing(as it's just getting a list of the 'connected devices' and the various addresses that are related to it), I just can't control what the actual implementation returns in this instance.